app/frontend/wallet/blockchain-explorer.ts

```diff
@@ -360,17 +361,28 @@ const blockchainExplorer = (ADALITE_CONFIG) => {
     validStakepools: ValidStakePoolsMapping,
     epochsToRewardDistribution: number
   ): Promise<NextRewardDetailsFormatted> {
+    const getPool = async (
+      poolHash: string
+    ): Promise<StakePoolInfo | HostedPoolMetadata | string> => {
```

Comment:
this type ambiguity seems very confusing, a proper fix would probably involve significantly more changes/refactoring, so at least I explicitly put here the types to "document" the current behavior